operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/BulkDependentResourceReconciler.java

```diff
@@ -0,0 +1,122 @@
+    desiredResources.forEach((key, value) -> {
+      final var instance =
+          updatable ? new UpdatableBulkDependentResourceInstance<>(bulkDependentResource, value)
+              : new BulkDependentResourceInstance<>(bulkDependentResource, value);
+      results.add(instance.reconcile(primary, actualResources.get(key), context));
+    });
+
+    return ReconcileResult.aggregatedResult(results);
+  }
+
+  @Override
+  public void delete(P primary, Context<P> context) {
+    var actualResources = bulkDependentResource.getSecondaryResources(primary, context);
+    deleteBulkResourcesIfRequired(Collections.emptySet(), actualResources, primary, context);
+  }
+
+  protected void deleteBulkResourcesIfRequired(Set<String> expectedKeys,
+      Map<String, R> actualResources, P primary, Context<P> context) {
+    actualResources.forEach((key, value) -> {
+      if (!expectedKeys.contains(key)) {
+        bulkDependentResource.deleteBulkResource(primary, value, key, context);
+      }
+    });
+  }
+
+  private static class BulkDependentResourceInstance<R, P extends HasMetadata>
+      extends AbstractDependentResource<R, P>
+      implements Creator<R, P>, Deleter<P> {
+    private final BulkDependentResource<R, P> bulkDependentResource;
+    private final R desired;
+
+    private BulkDependentResourceInstance(BulkDependentResource<R, P> bulkDependentResource,
+        R desired) {
+      this.bulkDependentResource = bulkDependentResource;
+      this.desired = desired;
+    }
+
+    @SuppressWarnings("unchecked")
+    private AbstractDependentResource<R, P> asAbstractDependentResource() {
+      return (AbstractDependentResource<R, P>) bulkDependentResource;
+    }
+
+    @Override
+    protected R desired(P primary, Context<P> context) {
+      return desired;
+    }
+
+    @SuppressWarnings("unchecked")
+    public R update(R actual, R desired, P primary, Context<P> context) {
+      return ((Updater<R, P>) bulkDependentResource).update(actual, desired, primary, context);
+    }
+
+    @Override
+    public Result<R> match(R resource, P primary, Context<P> context) {
+      return bulkDependentResource.match(resource, desired, primary, context);
+    }
+
+    @Override
+    protected void onCreated(ResourceID primaryResourceId, R created) {
+      asAbstractDependentResource().onCreated(primaryResourceId, created);
+    }
+
+    @Override
+    protected void onUpdated(ResourceID primaryResourceId, R updated, R actual) {
+      asAbstractDependentResource().onUpdated(primaryResourceId, updated, actual);
+    }
+
+    @Override
+    public Class<R> resourceType() {
+      return asAbstractDependentResource().resourceType();
+    }
+
+    @Override
+    public R create(R desired, P primary, Context<P> context) {
+      return bulkDependentResource.create(desired, primary, context);
+    }
+  }
+
+  private static class UpdatableBulkDependentResourceInstance<R, P extends HasMetadata>
```

Comment:
I don't get why this is needed.

Reply:
This is needed so that we can have the exact same interface for a "single" or bulk dependent in `AbstractDependentResource`.